### PR TITLE
[data] [streaming] Improve progress bar rendering on large scale jobs

### PR DESCRIPTION
**Pull request by ericl**
## Why are these changes needed?

For large jobs, the progress bar can be jerky since we do not limit the number of events processed per progress bar update.

### CODE REVIEW

File: python/ray/data/dataset.py [hunk 1]
```diff
@@ -405,17 +405,6 @@ def map_batches(
         To learn more about writing functions for :meth:`~Dataset.map_batches`, read
         :ref:`writing user-defined functions <transform_datasets_writing_udfs>`.
 
-        .. tip::
```
Comment (ericl) [inline python/ray/data/dataset.py:408]: Remove overly spammy / incorrect tips

Comment (jianoaix) [reply python/ray/data/dataset.py:408]: Why it's incorrect though? We no longer want to move usage to higher-level APIs like preprocessor ?

**Review by jianoaix — APPROVED** [review]

File: python/ray/data/dataset.py (hunk 1 above)
Comment (ericl) [reply python/ray/data/dataset.py:408]: It's kind of the opposite of https://github.com/ray-project/enhancements/pull/25.

I don't think this was ever correct in the first place. Also, it's not appropriate for Data users to be told to use preprocessor when they may not need it.

**Comment by ericl** [comment]
@zhe-thoughts we could consider merging this for 2.4, don't have a strong opinion.

**Review by c21 — APPROVED** [review]

**Comment by zhe-thoughts** [comment]
OK then let's wait until we unfreeze master.

**Pull request merged**